spectator/writer/writer.go

```diff
@@ -32,6 +32,12 @@ func (m *MemoryWriter) Lines() []string {
 	return slices.Clone(m.lines)
 }
 
+func (m *MemoryWriter) Reset() {
+	m.mu.RLock()
```

Comment:
Whoops, I should have looked closer. You want to use `m.mu.Lock()` (and `Unlock()`) later. `RLock/RUnlock` are for readers.

Reply:
Fixed.